x-pack/elastic-agent/pkg/core/monitoring/server/process.go

```diff
@@ -67,8 +67,14 @@ func processHandler(statsHandler func(http.ResponseWriter, *http.Request) error)
 			// proxy stats for elastic agent process
 			return statsHandler(w, r)
 		}
+		// TODO: allowlist of accepted endpoints to proxy to?
+		beatsEndpoint := vars["beatsEndpoint"]
```

Comment:
I'm not sure if subprocesses will mostly be beats, so we can limit the allowed paths to `/`, `/state`, and `/stats`, or how you want to handle this.

Reply:
definitely add allowlist here, including some proper escaping

Reply:
what escaping are you looking for? I added just a direct match check for `/`, `/state`, and `/stats`, so I'm not sure what to escape